### PR TITLE
BLS aggregation of 2 signers

### PR DESCRIPTION
**Pull request by 0xRampey**
fixes https://github.com/stackrlabs/succinct-residency-exps/issues/50

### CODE REVIEW

**Comment by 0xRampey** [comment]
`native_jolt` and `wasm_jolt` are OOMing
Need to increase image size and try again with Delphinus

File: bls-agg/rust/wasm/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+use thiserror::Error;
+
+#[no_mangle]
+pub fn bls_aggregate(num_signers: u32) -> u32 {
+    let private_keys: Vec<_> = (0..num_signers)
+            .map(|i| PrivateKey::new(&[i as u8; 32]))
+            .collect();
+
+        // generate messages
+        let messages: Vec<Vec<u8>> = (0..num_signers)
```
Comment (aashutoshrathi) [inline bls-agg/rust/wasm/src/lib.rs:16]: let's just generate one message and sign it from all

**Review by aashutoshrathi — CHANGES_REQUESTED** [review]
rest LGTM

File: bls-agg/rust/wasm_sp1/script/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        proof
+            .save("proof-with-pis.bin")
+            .expect("saving proof failed");
+
+
+        println!("Successfully generated proof!");
+
+        proof
+        .save("proof-with-pis.bin")
+        .expect("saving proof failed");
```
Comment (aashutoshrathi) [inline bls-agg/rust/wasm_sp1/script/src/main.rs:68]:
```suggestion
        let proof_bytes = proof.bytes();
        println!("Proof Size: {}", proof_bytes.len());
```

File: bls-agg/rust/wasm_risc_zero/host/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,47 @@
+// These constants represent the RISC-V ELF and the image ID generated by risc0-build.
+// The ELF is used for proving and the ID is used for verification.
+use methods::{GUEST_CODE_FOR_MERKEL_TREE_ELF, GUEST_CODE_FOR_MERKEL_TREE_ID};
```
Comment (aashutoshrathi) [inline bls-agg/rust/wasm_risc_zero/host/src/main.rs:3]: nit: let's change these vars

File: bls-agg/rust/native_sp1/script/src/main.rs [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+        proof
+        .save("proof-with-pis.bin")
+        .expect("saving proof failed");
+    
+        println!("Successfully generated proof!");
+
+        proof
+        .save("proof-with-pis.bin")
+        .expect("saving proof failed");
```
Comment (aashutoshrathi) [inline bls-agg/rust/native_sp1/script/src/main.rs:64]: let's just do
```suggestion
        let proof_bytes = proof.bytes();
        println!("Proof Size: {}", proof_bytes.len());
```

File: bls-agg/rust/native_risc_zero/host/src/main.rs [hunk 5]
```diff
@@ -0,0 +1,48 @@
+// These constants represent the RISC-V ELF and the image ID generated by risc0-build.
+// The ELF is used for proving and the ID is used for verification.
+use methods::{GUEST_CODE_FOR_MERKELIZATION_ELF, GUEST_CODE_FOR_MERKELIZATION_ID};
```
Comment (aashutoshrathi) [inline bls-agg/rust/native_risc_zero/host/src/main.rs:3]: nit: change the vars

**Review by aashutoshrathi — CHANGES_REQUESTED** [review]

**Review by aashutoshrathi — APPROVED** [review]

**Pull request merged**